### PR TITLE
[General] Max amount to roll

### LINKED ISSUE #3284: Cannot answer to roll
# Command bugs

#### Command name

`roll`.

#### What cog is this command from?

From : `General`.

#### What were you expecting to happen?

The command should answer the number.

#### What actually happened?

The command return an error message. `Command raised an exception: HTTPException: 400 BAD REQUEST (error code: 50035): Invalid Form Body
In content: Must be 2000 or fewer in length.`

#### How can we reproduce this issue?

You can use this command : 

`.roll 99990123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789`

The bot CAN answer to this, it just need to remove the author mention and the 2 🎲

### PR DESCRIPTION
**Pull request by Stonedestroyer**
### Type

- [X] Bugfix
- [X] Enhancement
- [ ] New feature

### Description of the changes

Resolves https://github.com/Cog-Creators/Red-DiscordBot/issues/3284. Not sure if the if statements were the best idea.

### CODE REVIEW

File: redbot/cogs/general/general.py [hunk 1]
```diff
@@ -87,15 +87,22 @@ def __init__(self):
...
             n = randint(1, number)
             await ctx.send(
                 "{author.mention} :game_die: {n} :game_die:".format(
                     author=author, n=humanize_number(n)
                 )
             )
-        else:
+        elif number <= 1:
             await ctx.send(_("{author.mention} Maybe higher than 1? ;P").format(author=author))
+        elif number > maxint:
```
Comment (Flame442) [inline redbot/cogs/general/general.py]: This should:tm: always be True, so just an `else` would technically work.

File: redbot/cogs/general/general.py [hunk 2]
```diff
@@ -87,15 +87,22 @@ def __init__(self):
...
+            await ctx.send(
+                _("{author.mention} Max allowed number is {maxamount}.").format(
+                    author=author, maxamount=maxint
```
Comment (Flame442) [inline redbot/cogs/general/general.py]: The `maxint` might look better if it is also passed to `humanize_number`.

**Review by Flame442 — DISMISSED** [review]
Neither of these are required, but they would be nice.

**Comment by Stonedestroyer** [comment]
I have implemented the QoL changes you suggested.

**Review by Flame442 — APPROVED** [review]

**Pull request merged**